owners/src/reviewer_selection.js

```diff
@@ -0,0 +1,186 @@
+/**
+ * Copyright 2019 The AMP HTML Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+
+=== Reviewer Selection Algorithm ===
+
+Inputs: a map from files needing approval to the nearest ownership subtree
+
+1. Select the set of ownership rules at the greatest depth in the tree (referred
+   to as "deepest rules" going forward) and union all owners; this is the set of
+   potential reviewers.
+2. For each potential reviewer, count the number of files they can approve, and
+   select whichever potential reviewer could cover the most additional files.
+3. Remove all files from the set that can be approved by the new reviewer, and
+   all ownership rules which are fully satisfied/which contain the new reviewer.
+
+Repeat until all files have an OWNER in the reviewer set.
+
+*/
+
+/**
+ * Class implementing the reviewer selection algorithm.
+ */
+class ReviewerSelection {
+  /** Part 1 **/
```

Comment:
I find this extremely amusing 😝

Reply:
Lol yeah probably not strictly necessary but :man_shrugging: